web/src/compose_actions.js

```diff
@@ -70,6 +71,28 @@ function show_compose_box(msg_type, opts) {
     // When changing this, edit the 42px in _maybe_autoscroll
     $(".new_message_textarea").css("min-height", "3em");
     compose_ui.set_focus(msg_type, opts);
+    const filter = narrow_state.filter();
+    if (filter && !filter.is_common_narrow()) {
```

Comment:
You need `if (filter && !filter.is_search()) {` condition here. `!filter.is_common_narrow` will trigger it for most of the narrows.

Reply:
The condition with `!filter.is_search()` doesn't work for triggering the banner, so I left it as `!filter.is_common_narrow()`.

Reply:
Oh! sorry I meant `if (filter && filter.is_search()) {`. You shouldn't just use any condition that works.

Reply:
Thanks, fixed!